components/nimbus/src/stateful/nimbus_client.rs

```diff
@@ -704,6 +719,23 @@ impl NimbusClient {
         }
         Ok(())
     }
+
+    #[cfg(test)]
+    pub fn get_metrics_handler(&self) -> &TestMetrics {
+        let metrics = &*self.metrics_handler;
+        unsafe { std::mem::transmute::<&Box<dyn MetricsHandler>, &Box<TestMetrics>>(metrics) }
+    }
+
+    #[cfg(test)]
+    pub fn get_recorded_context(&self) -> Option<&Arc<TestRecordedContext>> {
+        self.recorded_context
+            .clone()
+            .map(|ref recorded_context| unsafe {
+                std::mem::transmute::<&Arc<dyn RecordedContext>, &Arc<TestRecordedContext>>(
+                    recorded_context,
+                )
+            })
```

Comment:
I'm pretty sure this is wrong because `self.recorded_context` is an `Option<JsonObject>`.